app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,60 @@
+class CartsController < ApplicationController
+  before_action :load_food, only: :create
+
+  def index
+    load_cart
+  end
+
+  def create
+    @food_id = params[:food_id].to_i
+    @quantity = params[:quantity].to_i
+    if @quantity.positive?
+      if food_in_cart?
+        @quantity = session[:cart][@food_id.to_s] + @quantity
+        check_quantity
+      end
+    else
+      @quantity = 1
+    end
+    session[:cart][@food_id.to_s] = @quantity
+    load_cart
+  end
+
+  def destroy
+    food_id = params[:id].to_i
+    session[:cart].delete(food_id.to_s)
+    load_cart
+  end
+
+  private
+
+  def load_food
+    @food = Food.find_by id: params[:food_id]
+    return if @food
+
+    flash[:danger] = t "show_food_fail"
+    redirect_to root_path
+  end
+
+  def load_cart
+    qtity = session[:cart]
+    @carts = Food.find_foods_cart(qtity.keys)
```

Comment:
Tìm ko thấy thì sao em ơi

Reply:
em có kiểm tra bên view nếu carts trống thì in ra ko có sản phẩm nào trong giỏ hàng